lib/Hakyll/Core/Rules.hs

```diff
@@ -1,19 +1,32 @@
 --------------------------------------------------------------------------------
--- | This module provides a declarative DSL in which the user can specify the
--- different rules used to run the compilers.
---
--- The convention is to just list all items in the 'Rules' monad, routes and
--- compilation rules.
---
--- A typical usage example would be:
---
--- > main = hakyll $ do
--- >     match "posts/*" $ do
--- >         route   (setExtension "html")
--- >         compile someCompiler
--- >     match "css/*" $ do
--- >         route   idRoute
--- >         compile compressCssCompiler
+{- | This module provides a declarative Domain Specific Language (DSL) to generate a static site
+by specifying transformation 'Rules' (although the use case is not limited to static sites).
+__Each rule is a processing pipeline__ which normally consists of three parts:
+
+1. Inputs (like Markdown files) to process (collected with e.g. 'match' or 'create').
+2. Transformation steps (like Markdown to HTML) to compute ("compile") for each input its output content
+(collected with 'compile' and 'Hakyll.Core.Compiler.Compiler').
+3. Routing to determine to which file an output content will be written out. For a static site
+this translates to determine under which URL the output content will be accessible
+(configured with 'route' and 'Hakyll.Core.Routes.Routes').
+
+A typical usage example looks as follows:
+
+> {-# LANGUAGE OverloadedStrings #-}    -- write 'match "posts/**.md"' instead of 'match $ fromGlob "posts/**.md"'
```

Comment:
I can't get this extension to show up in the docs. Any ideas?

Reply:
Apparently one can't nest block comments inside Haddock block comments. You'll have to switch to outer one:

```
-- | Blah blah
--
-- > {-# LANGUAGE OverloadedStrings #-}
```

Reply:
Thanks. I added the workaround. Here's the corresponding bug ticket for Haddock https://github.com/haskell/haddock/issues/1254